File: Merge Sorted Array.py
#Time Complexity: O(n)
#Space Complexity: O(1)

class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        firstPointer=m-1
        secondPointer=m+n-1
        thirdPointer=n-1
        while firstPointer>=0 and thirdPointer>=0:
            if nums2[thirdPointer]>nums1[firstPointer]:
                nums1[secondPointer]=nums2[thirdPointer];
                
                thirdPointer-=1;
            else:
                nums1[secondPointer]=nums1[firstPointer];
                firstPointer-=1;
            secondPointer-=1
        if thirdPointer>=0:
            nums1[:thirdPointer+1]=nums2[:thirdPointer+1];
        

File: MergeSortedArray.java
//Time Complexity: O(m + n)
//Space Complexity: O(1)
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int num1Pointer = m -1;
        int num2Pointer = n - 1;
        int index = m + n -1;
        while(num1Pointer >= 0 && num2Pointer >= 0) {
            if(nums1[num1Pointer] >= nums2[num2Pointer]) {
                nums1[index] = nums1[num1Pointer];
                num1Pointer--;

            }
            else {
                nums1[index] = nums2[num2Pointer];
                num2Pointer--;

            }
            index--;
        }
        while(num2Pointer >= 0) {
            nums1[index] = nums2[num2Pointer];
            index--;
            num2Pointer--;
        }


    }
}